packages/notary/src/notary/notary.service.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    InternalServerErrorException,
+    Injectable,
+    OnModuleInit,
+    NotFoundException
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { ethers } from 'hardhat';
+import { providers } from 'ethers';
+
+import { Notary, Notary__factory } from '~/typechain';
+import { NotaryContract } from './notary-contract.entity';
+
+@Injectable()
+export class NotaryService implements OnModuleInit {
+    constructor(
+        @InjectRepository(NotaryContract)
+        private readonly repository: Repository<NotaryContract>
+    ) {}
+
+    async onModuleInit(): Promise<void> {
+        try {
+            await this.getNotaryContract();
+        } catch (e) {
+            if (e instanceof NotFoundException) {
+                await this.deploy();
+                return;
+            }
+
+            throw e;
+        }
+    }
+
+    async getNotaryContract(): Promise<NotaryContract> {
+        const contracts = await this.repository.find();
+
+        if (contracts.length < 1) {
+            throw new NotFoundException(`Notary contract not deployed`);
+        }
+
+        return contracts[0];
+    }
```

Comment:
Wouldn't it be better to return **NotaryContract** or **null** from **getNotaryContract()** and check for null instead of try/catch? Especially since it's not exceptional that there is no contract (at least I think so).

Reply:
It's not exceptional during the module init, but this method is also used during the execution of the app.
And during the execution it should never fail, that's why I throw.